### ISSUE TITLE
Gmail's signature is not being detected

### ISSUE THREAD

**Issue by sashkab** [github/email_reply_parser#46]
When you parse emails, you detect signature by  `-- ` (dash-dash-space). But, Gmail  for some odd reason, adds asterisks before and after signature delimiter, you should, probably, support this as well. In raw email it looks like this: `*-- *`.

**Comment by crimsonvspurple** [comment]
Any love for this?

**Comment by peter-evans** [comment]
👋 @sashkab @crimsonvspurple 

I realise this is a very old issue, but if this is still affecting folks I would like to hear more detail because I've been unable to reproduce this. As far as I can tell, Gmail does use the [standard signature delimiter](https://en.wikipedia.org/wiki/Signature_block#Standard_delimiter), `-- \n`.

**Comment by sashkab** [comment]
I've reported this almost ten years ago, I no longer have access to the systems I used to run this on, and email accounts, so I cannot reproduce this problem. 

Looks like gmail doesn't add `-- ` automatically prior to signature, I have to add, and if I do, and mark text as bold, I will end up with `*-- *` instead. 

<img width="362" alt="Image" src="https://github.com/user-attachments/assets/cf9c04d7-3243-4c3b-a422-fead9cefe9e9" />

Unable to verify with the original message, as I no longer have access to that email account. Well, it was almost ten years ago :)

**Comment by peter-evans** [comment]
I appreciate it was a long time ago! Thank you for replying.

> Looks like gmail doesn't add `-- ` automatically prior to signature, I have to add, and if I do, and mark text as bold, I will end up with `*-- *` instead.

When you say that you mark the text as bold, are you selecting the `-- `, too? Or just the text underneath the delimiter?

I just tried it by only marking the text under the delimiter as bold, and it correctly renders the raw email when sent.

```
--
*my sig*
```

If you are selecting the delimiter, too, then that is the cause and there is not much that can be done about that. The delimiter must be `-- \n` to conform to [the standard](https://en.wikipedia.org/wiki/Signature_block#Standard_delimiter) used by all mail client/servers.

**Comment by sashkab** [comment]
Obviously, in this case I'm selecting `-- ` as bold to demonstrate the issue right now. I don't think I selected it in the past, but who knows what was happening. I'm surprised, Google doesn't automatically adds `-- ` prior to signature right now, perhaps in the past they did. 

My point is, no matter how smart the program is, there are always users, who manage to break things. While I don't think adding `*-- *` delimiter is correct thing to do, it might be a way to kill a class of failures of the module.
